src/widget/whotcuebutton.cpp

```diff
@@ -129,16 +145,95 @@ void WHotcueButton::mousePressEvent(QMouseEvent* e) {
     }
 
     // Pass all other press events to the base class.
-    WPushButton::mousePressEvent(e);
+    WPushButton::mousePressEvent(pEvent);
 }
 
-void WHotcueButton::mouseReleaseEvent(QMouseEvent* e) {
-    const bool rightClick = e->button() == Qt::RightButton;
+void WHotcueButton::mouseReleaseEvent(QMouseEvent* pEvent) {
+    const bool rightClick = pEvent->button() == Qt::RightButton;
     if (rightClick) {
         // Don't handle stray release events
         return;
     }
-    WPushButton::mouseReleaseEvent(e);
+    WPushButton::mouseReleaseEvent(pEvent);
+}
+
+void WHotcueButton::mouseMoveEvent(QMouseEvent* pEvent) {
+    TrackPointer pTrack = PlayerInfo::instance().getTrackInfo(m_group);
+    if (!pTrack) {
+        return;
+    }
+    // Only allow moving set hotcues to empty or set slots.
+    // Note that Track::swapHotcues() allows both directions.
+    if (m_hotcue == Cue::kNoHotCue) {
+        return;
+    }
+
+    if (DragAndDropHelper::mouseMoveInitiatesDrag(pEvent)) {
+        QDrag* pDrag = new QDrag(this);
+        QMimeData* mimeData = new QMimeData;
+        mimeData->setText(mimeTextIdentifier() + kMimeTextDelimiter + QString::number(m_hotcue));
+        pDrag->setMimeData(mimeData);
```

Comment:
this is slightly safer because it doesn't leak in case something throws before calling `setMimeData`. Also avoids a "naked new".
```suggestion
        auto mimeData = std::make_unique<QMimeData>();
        mimeData->setText(mimeTextIdentifier() + kMimeTextDelimiter + QString::number(m_hotcue));
        pDrag->setMimeData(mimeData.release());
```